### PR TITLE
fix: updated documentation for trestle

### PR DESCRIPTION
**Pull request by vikas-agarwal76**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [ ] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [ ] My PR meets testing requirements.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -62,11 +63,13 @@ which appears, for a catalog a user decides is titled nist-800-53, as:
 
 ```
 
-`dist` directory will contain the assembled version of the models located on the source model directories (at the project root level) which are: `catalogs`, `profiles`, `component-definitions`, `system-security-plans`, `assessment-plans`, `assessment-results` and `plan-of-action-and-milestones`. The assumption is that each of the OSCAL files within this folder is ready to be read by external 3rd party tools.
+`dist` directory will contain the assembled version of the top level models located on the source model directories which are: `catalogs`, `profiles`, `component-definitions`, `system-security-plans`, `assessment-plans`, `assessment-results` and `plan-of-action-and-milestones`. The assumption is that each of the OSCAL files within this folder is ready to be read by external 3rd party tools.
```
Comment (fsuits) [inline docs/cli.md:66]: I would remove this since it is repeated just below here.  I would keep this section high level.

File: docs/cli.md [hunk 2]
```diff
@@ -62,11 +63,13 @@ which appears, for a catalog a user decides is titled nist-800-53, as:
...
+
+In most of the places in the documentation we use `json` format for specifying model files, but they are equally applicable to `yaml` format also.
 
 ### Support for subdivided document structures
 
-The files constructed by OSCAL can run into the tens of thousands of lines of yaml or formatted json. At this size the
+The files constructed by OSCAL can run into tens of thousands of lines of yaml or formatted json. At this size the
 files become completely unmanageable for users. To combat this, trestle can `trestle split` a file into many smaller files and later merge those split files together.
```
Comment (fsuits) [inline docs/cli.md:73]: This may be a good place to mention that priority is to json and yaml is supported as best effort.

File: docs/cli.md [hunk 3]
```diff
@@ -193,7 +197,7 @@ The following options are supported:
 The user can edit the parts of the generated OSCAL model by modifying the sample content in those directories.
 
 The initial level of decomposition of each type of model varies according to the model type.
-This default or reference decomposition behaviour can be changed by modifying the rules in a `.trestle/config file`. These rules can be written as a sequence of `trestle split` commands.
+This default or reference decomposition behaviour can be changed by modifying the rules in `.trestle/config.ini` file. These rules can be written as a sequence of `trestle split` commands.
```
Comment (fsuits) [inline docs/cli.md:200]: I think we want to remove all references to config.ini

File: docs/cli.md [hunk 4]
```diff
@@ -240,7 +244,7 @@ If the element is of JSON/YAML type array list and you want trestle to create a
 
 If you just want to split a file into all its constituent parts and the file does not contain a simple list of objects, you can still use `*` and the file will be split into all its non-trivial elements.  Thus if you split a catalog with `-e catalog.*` the result will be a new directory, `catalog`, containing files representing the large items, `back-matter.json, groups.json and metadata.json`, but there will still be a `catalog.json` file containing just the catalog's `uuid`.  Small items such as strings and dates cannot be split off and will remain in the original model file that is being split.
 
-Here are some examples.  Starting with a single catalog file, `my_catalog/catalog.json`, if we do `trestle split -f catalog.json -e 'catalog.*'` we end up with:
+Here are some examples.  Starting with a single catalog file, `my_catalog/catalog.json`, if I do `trestle split -f catalog.json -e 'catalog.*'` I end up with:
```
Comment (fsuits) [inline docs/cli.md:247]: We may need to check all the docs. to convert we to I if it is done elsewhere like this.

File: docs/cli.md [hunk 5]
```diff
@@ -361,14 +363,13 @@ Note that the type of the file is now `stripped.Catalog` and it no longer contai
 
 ## `trestle partial-object-validate`
 
-OSCAL objects are extremely large. Some systems may only be able to produce partial OSCAL objects. For example
-the tanium-to-oscal task produces the `results` attribute of an `assessment-results` object.
+OSCAL objects are extremely large. Some systems may only be able to produce partial OSCAL objects. For example, the tanium-to-oscal task produces the `results` attribute of an `assessment-results` object.
```
Comment (fsuits) [inline docs/cli.md:367]: I would say OSCAL objects *may* be extremely large.

File: docs/cli.md [hunk 6]
```diff
@@ -298,9 +302,7 @@ The following option is required:
...
-split from a component, `.*` can be used. For example, `trestle merge -e 'catalog.*'` command, issued from the directory where `catalog.json` or`catalog` directory exists, will merge every single sub-component of that catalog back into the `catalog.json`.
+For example, in the command `trestle merge -e 'catalog.metadata'`, executed in the same directory where `catalog.json | catalog.yaml` or the split `catalog` directory exists, the property `metadata` from `metadata.json` or `metadata.yaml` would be moved/merged into `catalog.json` or `catalog.yaml`. If the `metadata` model has already been split into smaller sub-component models previously, those smaller sub-components are first recusively merged into `metadata`, before merging `metadata` subcomponent into `catalog`. To specify merging every sub-component split from a component, `.*` can be used. For example, `trestle merge -e 'catalog.*'` command, issued from the directory where `catalog.json` or`catalog` directory exists, will merge every single sub-component of that catalog back into the `catalog.json`.
```
Comment (fsuits) [inline docs/cli.md:305]: I don't recommend listing both json and yaml options everywhere - it makes it harder to read.  I would just use json but occasionally remind people yaml is also possible.

File: docs/contributing/website.md [hunk 7]
```diff
@@ -13,6 +13,8 @@ The current documentation website build reuses some of these files, specifically
 - Entirety of `CONTRIBUTING.md`
 - Entirety of `CODE_OF_CONDUCT.md`
 - Entirety of `CHANGELOG.md`
+- Entirely of `MAINTAINERS.md`
+- Entirely of `DCO1.1.txt`
```
Comment (fsuits) [inline docs/contributing/website.md:17]: Please confirm you wanted to add these here.  I thought you had found a reason they should be separate but I may have misunderstood.

File: docs/trestle_author.md [hunk 8]
```diff
@@ -170,7 +170,7 @@ version _of that document_ (be it an instance or the template itself). This mean
 
 ## `trestle author folders`
 
-\`author folders is designed to allow the assembly of groups of templates where the folder assembly is the unique instance. Trestle author folders supports validation of both markdown and drawio files. Note that headers / metadata must be specified in each applicable template.
+\`author folders\` is designed to allow the assembly of groups of templates where the folder assembly is the unique instance. Trestle author folders supports validation of both markdown and drawio files. Note that headers / metadata must be specified in each applicable template.
```
Comment (fsuits) [inline docs/trestle_author.md:174]: I think the slashes are not needed here - just the backticks.

**Review by fsuits — CHANGES_REQUESTED** [review]
Looks good - please review the suggested changes.

File: docs/cli.md (hunk 1 above)
Comment (vikas-agarwal76) [reply docs/cli.md:66]: done.

File: docs/cli.md (hunk 2 above)
Comment (vikas-agarwal76) [reply docs/cli.md:73]: specified above when talking about 2 formats.

File: docs/cli.md (hunk 3 above)
Comment (vikas-agarwal76) [reply docs/cli.md:200]: Removed this para completely as during create only one .json file is created and not a split model.

File: docs/cli.md (hunk 4 above)
Comment (vikas-agarwal76) [reply docs/cli.md:247]: checked it here.

File: docs/cli.md (hunk 5 above)
Comment (vikas-agarwal76) [reply docs/cli.md:367]: done

File: docs/cli.md (hunk 6 above)
Comment (vikas-agarwal76) [reply docs/cli.md:305]: done

File: docs/contributing/website.md (hunk 7 above)
Comment (vikas-agarwal76) [reply docs/contributing/website.md:17]: These are those files which are used from the main github project directory. The markdown files are directly included in the corresponding files in docs directory, while the LICENSE and DOC and text files and are include using docs automation script to convert them to markdown format.

File: docs/trestle_author.md (hunk 8 above)
Comment (vikas-agarwal76) [reply docs/trestle_author.md:174]: done.

**Review by fsuits — APPROVED** [review]
LGTM

**Pull request merged**